### PR TITLE
Speedup some unnecessarily slow tests

### PR DESCRIPTION
**Pull request by gassmoeller**
This tries to address #3628. On my system (limited a docker container to 4 threads) this bring some substantial benefits:

before: 5116.58
after:  3951.27

Lets see what the tester say (I will start it on purpose on one of the slow testers).

### CODE REVIEW

File: tests/discontinuous_temperature_blankenbach/screen-output [hunk 1]
```diff
@@ -139,10242 +139,18 @@ Number of degrees of freedom: 1,235 (578+81+576)
...
-     Heating rate (average/total):       0 W/kg, 0 W
-
-*** Timestep 587:  t=0.102547 seconds
-   Solving temperature system... 4 iterations.
-   Rebuilding Stokes preconditioner...
-   Solving Stokes system... 3+0 iterations.
-
-   Postprocessing:
-     RMS, max velocity:                  895 m/s, 1.73e+03 m/s
-     Temperature min/avg/max:            -0.006475 K, 0.5 K, 1.
```
Comment (tjhei) [inline tests/discontinuous_temperature_blankenbach/screen-output:10366]: We were doing 939 timesteps? Wow.

**Comment by tjhei** [comment]
This looks great. One of the tests is failing though.

How did you pick the list of tests? Would it be worth sorting all tests by time and take a look at the slowest?

**Comment by gassmoeller** [comment]
> One of the tests is failing though.

That is on me, I forgot to update after I made a final change. I updated the results.

> How did you pick the list of tests? Would it be worth sorting all tests by time and take a look at the slowest?

That is essentially what I did. I took a look at tests that took more than roughly 100-150 seconds and selected the ones where I could figure out the meaning of the test and where I thought I could make them faster without removing the purpose of the test. I think we now have no tests left that take more than 200 seconds, and most between 100-200 are related to melt solvers or 3D models where the test might not converge with lower resolution or make no sense with shorter end time. There are a bunch of tests around 100 seconds for the Newton solver that I think could be simplified, but the solver settings are a bit tricky so I left them alone.

**Comment by gassmoeller** [comment]
No luck, while the tests are faster the slow testers still timeout (just barely, they timeout after finishing all tests and running through cmake). I restarted the run on a fast tester to at least get this passed and merged.

**Comment by tjhei** [comment]
This looks great, so I will merge.

**Pull request merged**